roachprod: Add support for io2 SSD for AWS

miretskiy opened the pull request:
Support io2 SSDs for aws.
Support aws-image-ami override.

Release Notes: None
Release Justification: No user changes; N/A

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54774)
<!-- Reviewable:end -->

miretskiy commented:
bors r+

Commit:
roachprod: Add ability to specify image to use when provisioning.

Add a flag to override AWS image AMI when provisioning roachprod
cluster.

Add io2 support for AWS.

Release Notes: None
Release Justification: N/A; roachprod changes.

miretskiy commented:
bors r+
